### PR TITLE
Add ability to cross-link between documentation groups

### LINKED ISSUE #17: Cross module linking
Currently there's no way to cross-link modules, which would be handy. Perhaps this could be done via a custom URL scheme, as suggested by @rmccue.

### PR DESCRIPTION
**Pull request by rmccue**
Fixes #17.

I've added a helper function to generate URLs, which is filterable, as we want to generate URLs differently on the public site (cc @joehoyle).

Also adds documentation for this feature, and a stub page for the base of the documentation.

### CODE REVIEW

File: inc/class-markdownparser.php [hunk 1]
```diff
@@ -58,7 +62,29 @@ protected function inlineLink( $data ) {
 
 		// Override href.
 		$slug = get_slug_from_path( $root, $resolved );
-		$url = add_query_arg( [ 'id' => urlencode( $slug ) ] );
+		$url = get_url_for_page( UI\get_current_group_id(), $slug );
```
Comment (rmccue) [inline inc/class-markdownparser.php]: Ideally, this should pull the group ID from the page instead, but we're not passing this around with the page currently.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Documentation
```
Comment (joehoyle) [inline docs/README.md]: Is this expected to actually show in the Documentation groups list too? Seems a bit meta if so!

Comment (rmccue) [reply docs/README.md]: Yep, it's a module that you get included, so makes sense to be documented (similar to Core).

**Comment by joehoyle** [comment]
Why would I want to use this as someone building projects on Platform though? Are we intentionally recommending developers use this method to add project level documentation?

**Comment by rmccue** [comment]
> Are we intentionally recommending developers use this method to add project level documentation?

Eventually, IMO yes, although there's no documentation on this yet.

**Comment by joehoyle** [comment]
Without that, I'm not really following why we would add this, just because it's a module. We have Core documentation, because it provides developer APIs, but currently this documentation is, well, useless for the context of our developer documentation.

**Pull request merged**